integration/e2e/chaincode_service_test.go

```diff
@@ -0,0 +1,175 @@
+		}
+		os.RemoveAll(testDir)
+	})
+
+	Describe("basic solo network with 2 orgs and external chaincode service", func() {
+		var (
+			datagramReader *DatagramReader
+			ccserver       ifrit.Process
+		)
+
+		BeforeEach(func() {
+			datagramReader = NewDatagramReader()
+			go datagramReader.Start()
+
+			network = nwo.New(nwo.BasicSoloWithChaincodeServers(&extcc), testDir, nil, StartPort(), components)
+
+			//add extcc builder
+			network.ExternalBuilders = append(network.ExternalBuilders, extbldr)
+
+			network.MetricsProvider = "statsd"
+			network.StatsdEndpoint = datagramReader.Address()
+			network.Profiles = append(network.Profiles, &nwo.Profile{
+				Name:          "TwoOrgsBaseProfileChannel",
+				Consortium:    "SampleConsortium",
+				Orderers:      []string{"orderer"},
+				Organizations: []string{"Org1", "Org2"},
+			})
+			network.Channels = append(network.Channels, &nwo.Channel{
+				Name:        "baseprofilechannel",
+				Profile:     "TwoOrgsBaseProfileChannel",
+				BaseProfile: "TwoOrgsOrdererGenesis",
+			})
+
+			network.GenerateConfigTree()
+
+			// package connection.json
+			extcc.CodeFiles = map[string]string{
+				network.ChaincodeConnectionPath(&extcc): "connection.json",
+			}
+
+			for _, peer := range network.PeersWithChannel("testchannel") {
+				core := network.ReadPeerConfig(peer)
+				core.VM = nil
+				network.WritePeerConfig(peer, core)
+			}
+			network.Bootstrap()
+
+			networkRunner := network.NetworkGroupRunner()
+			process = ifrit.Invoke(networkRunner)
+			Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+		})
+
+		AfterEach(func() {
+			if datagramReader != nil {
+				datagramReader.Close()
+			}
+			if ccserver != nil {
+				ccserver.Signal(syscall.SIGTERM)
+				Eventually(ccserver.Wait(), network.EventuallyTimeout).Should(Receive())
+			}
+		})
+
+		It("executes a basic solo network with 2 orgs and external chaincode service", func() {
+			By("getting the orderer by name")
+			orderer := network.Orderer("orderer")
+
+			By("setting up the channel")
+			network.CreateAndJoinChannel(orderer, "testchannel")
+			nwo.EnableCapabilities(network, "testchannel", "Application", "V2_0", orderer, network.Peer("Org1", "peer0"), network.Peer("Org2", "peer0"))
+
+			By("deploying the chaincode")
+			nwo.DeployChaincode(network, "testchannel", orderer, extcc)
+
+			By("starting the chaincode service")
+			extcc.SetPackageIDFromPackageFile()
+			address := network.ChaincodeServerAddress(&extcc)
+			Expect(address).NotTo(Equal(""))
+
+			// start external chaincode service
+			ccrunner := network.ChaincodeServerRunner(extcc, []string{extcc.PackageID, address})
+			ccserver = ifrit.Invoke(ccrunner)
+			Eventually(ccserver.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+			// init the chaincode, if required
+			if extcc.InitRequired {
+				By("initing the chaincode")
+				nwo.InitChaincode(network, "testchannel", orderer, extcc, network.PeersWithChannel("testchannel")...)
+			}
```

Comment:
This is hard-coded above.

Reply:
@sykesm Did not understand what is hardcoded between lines 150 and 153. Can you please elaborate?

Reply:
Line 137 calls `DeployChaincode`. This logic part of that function.

```
// DeployChaincode is a helper that will install chaincode to all peers that
// are connected to the specified channel, approve the chaincode on one of the
// peers of each organization in the network, commit the chaincode definition
// on the channel using one of the peers, and wait for the chaincode commit to
// complete on all of the peers. It uses the _lifecycle implementation.
// NOTE V2_0 capabilities must be enabled for this functionality to work.
func DeployChaincode(n *Network, channel string, orderer *Orderer, chaincode Chaincode, peers ...*Peer) {
	if len(peers) == 0 {
		peers = n.PeersWithChannel(channel)
	}
	if len(peers) == 0 {
		return
	}

	PackageAndInstallChaincode(n, chaincode, peers...)

	// approve for each org
	ApproveChaincodeForMyOrg(n, channel, orderer, chaincode, peers...)

	// commit definition
	CheckCommitReadinessUntilReady(n, channel, chaincode, n.PeerOrgs(), peers...)
	CommitChaincode(n, channel, orderer, chaincode, peers[0], peers...)

	// init the chaincode, if required
	if chaincode.InitRequired {
		InitChaincode(n, channel, orderer, chaincode, peers...)
	}
}
```

Reply:
@sykesm For chaincode as an external service, chaincode package ID is required to launch the chaincode. The package id is available after deployment of the chaincode. 
The DeployChaincode method has been updated with following just before chaincode initialization in this PR:
	if chaincode.Server {
		return
	}

After the chaincode is deployed, the package id is used to launch the chaincode process and it is then initialized - hence the init call outside DeployChaincode.
This appears to be acceptable solution.

Alternatively, since package id is hash of the chaincode package file - it can / be computed before deployment and chaincode process be launched prior to deploying the chaincode. In this way Init will succeed.
Please let me know this alternate path shall be explored.